File: app/src/main/java/com/harry/pullgo/TabButton.java
package com.harry.pullgo;

import android.content.Context;
import android.content.res.Resources;
import android.graphics.Canvas;
import android.graphics.Color;
import android.util.AttributeSet;
import android.view.MotionEvent;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.widget.AppCompatButton;
import androidx.constraintlayout.widget.ConstraintSet;

public class TabButton extends AppCompatButton {

    public TabButton(@NonNull Context context) {
        super(context);
        init(context);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
    }

    public TabButton(@NonNull Context context, @Nullable AttributeSet attrs) {
        super(context, attrs);
        init(context);
    }

    private void init(Context context){
        Resources resources=getResources();
        setBackgroundColor(Color.WHITE);
        setTextColor(resources.getColor(R.color.main_color));
    }
}

File: app/src/main/java/com/harry/pullgo/TabButton.kt
package com.harry.pullgo

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.util.AttributeSet
import androidx.appcompat.widget.AppCompatButton
import androidx.core.content.ContextCompat

class TabButton : AppCompatButton {
    constructor(context: Context) : super(context) {
        init(context)
    }

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
    }

    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        init(context)
    }

    private fun init(context: Context) {
        val resources = resources
        setBackgroundColor(Color.WHITE)
        setTextColor(ContextCompat.getColor(context,R.color.main_color))
    }
}